### PR TITLE
1240 android usability

### LINKED ISSUE #1240: Android usability
Some concerns about the android UI 
- on-screen keyboard

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #1240 

Improves the usability of the android app. There's one issue outstanding with printing, will raise a separate issue for that one.

The changes in this update are:
- update the login and initialise forms to allow for scrolling when the screen is small
- for the input elements on those forms, auto scroll into view on focus ( this brings the submit button into view too )
- resize modals if the screen size is too small to support the requested size
- for specific edit item modals, resize when an on-screen keyboard is displayed

Originally, when adding an item to an outbound shipment the view was:

add-item-landscape

With the changes, the view in portrait is usable:
add-item-portrait

and the landscape view is too:
add-item-landscape-new

If you'd like to test, here's an APK which includes the changes: https://drive.google.com/drive/u/2/folders/111xxY1rdSTxXwXfza9Y-kiVSEaSQcuCj

### CODE REVIEW

**Review by clemens-msupply — APPROVED** [review]
Looks good thanks!

Is it possible to not auto capitalize "Site name" and password in the bootstrap screen?

This PR is incomplete, right? e.g. Adding item modal for inbound shipments need similar change...

Once we got the reliable build system going we should remove the .so files from the repo, think it blows up the repo quite a bit...

**Comment by mark-prins** [comment]
Doh! sorry @clemens-msupply - have made the PR from the wrong branch 🙄 
This was to fix the language selection issue, and I've used the incomplete android UI branch. will push the rest of the changes and update this PR, and make a new one for the language..

File: packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEdit.tsx [hunk 1]
```diff
@@ -35,6 +36,7 @@ export const RequestLineEdit = ({
     useDraftRequisitionLine(currentItem);
   const { next, hasNext } = useNextRequestLine(currentItem);
   const nextDisabled = (!hasNext && mode === ModalMode.Update) || !currentItem;
+  const height = useKeyboardHeightAdjustment(600);
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEdit.tsx:39]: it was previously `600` - and yes, is different from the others. probably should be the same, but wanted to minimise the changes here!

File: packages/common/src/hooks/useKeyboardHeightAdjustment/useKeyboardHeightAdjustment.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { useEffect, useState } from 'react';
+import { Keyboard } from '@capacitor/keyboard';
+
+export const useKeyboardHeightAdjustment = (initialHeight: number) => {
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply packages/common/src/hooks/useKeyboardHeightAdjustment/useKeyboardHeightAdjustment.ts:4]: I've had to add in `@capacitor/core` and check for the plugin 😬

**Comment by mark-prins** [comment]
> Is it possible to not auto capitalize "Site name" and password in the bootstrap screen?

haven't been able to reproduce this! for me, neither field have autocapitalised 🤷 
However, have added the input hint to prevent it 😁 

> This PR is incomplete, right? e.g. Adding item modal for inbound shipments need similar change...

Not sure about this one either - the inbound shipment is already done

> Once we got the reliable build system going we should remove the .so files from the repo, think it blows up the repo quite a bit...

yes! this was just a helper really, so that f/e devs could build and test easily 😁

File: packages/common/src/hooks/useKeyboardHeightAdjustment/useKeyboardHeightAdjustment.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (clemens-msupply) [reply packages/common/src/hooks/useKeyboardHeightAdjustment/useKeyboardHeightAdjustment.ts:5]: Where is this check? can't find it...

**Comment by clemens-msupply** [comment]
Thanks autocapitalised is disabled now on my device

File: packages/common/src/hooks/useKeyboardHeightAdjustment/useKeyboardHeightAdjustment.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mark-prins) [reply packages/common/src/hooks/useKeyboardHeightAdjustment/useKeyboardHeightAdjustment.ts:5]: It's this line: https://github.com/openmsupply/openmsupply-client/blob/be084ef71a3f5e3826bd6f034933a3c4c56e5559/packages/common/src/hooks/useKeyboardHeightAdjustment/useKeyboardHeightAdjustment.ts#L9

which came in with this commit: be084ef

[the remark this answers is not in the document]
Comment (clemens-msupply) [reply packages/common/src/hooks/useKeyboardHeightAdjustment/useKeyboardHeightAdjustment.ts:5]: thanks, sorry was too obvious for me to find ;)

**Pull request merged**